lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+female, hindfoot length is 5 above the mean, and Dipodomys is yes?
+
+```{r, echo=FALSE, eval=FALSE}
+# Solution
+challenge1 <- portal2 %>% 
+    filter(taxa == "Rodent") %>% 
+    mutate(Dipodomys = ifelse(genus == "Dipodomys", "Yes", "No"))
+
+fit_challenge1 <- glm(weight ~ sex + hfl_mean_center + Dipodomys, data = challenge1)
+tidy(fit_challenge1)
+73 + (-0.027 * 0) + (5.499 * 5) + (-65.377 * 1)
+```
+
+2. Use the below code to create a new variable `Disease` with (random) values as
+either 0 for "Healthy" or 1 for "Diseased". Then, write up a formula using `glm`
+to analyze the role that weight, sex, and hindfoot length (*not* mean centered)
+has on disease status. Because the $y$ is not continuous, you need to
+set `family = binomial` in `glm`. Run the code, check the summary (or `tidy`),
+put the numbers into an equation (as we did above), and *try* to interpret the
+results (where sex is female, weight is 50, and hindfoot length is 30).
+
+```{r}
+# So random numbers is the same for everyone
+set.seed(1002)
+challenge2 <- portal %>% 
+    # rbinom randomly creates values for binary data.
+    mutate(Disease = rbinom(nrow(.), 1, 0.25))
+```
+
+```{r, echo=FALSE, eval=FALSE}
+# Solution
+fit_challenge2 <- glm(Disease ~ weight + sex + hindfoot_length, 
+                      data = challenge2, family = binomial)
+summary(fit_challenge2)
+tidy(fit_challenge2)
+-1.035 + (-0.0003 * 50) + (-0.0338 * 0) + (-0.0004 * 30)
+```
+
+### Logistic regression
+
+Logistic regression is a technique that has the $y$ as a binary or categorical
+variable (e.g. female vs male, mammal vs plant vs bacteria). We won't be
+covering this in too much detail, but I will go over it to highlight the
+flexibility that GLM has in handling different types of data.
+
+The reason why the result of the challenge two above was difficult to interpret
+was because there's a bit more to logistic regression then with linear
+regression. It becomes clearer when we look at the formula:
+
+$$ logit(p) = \ln\left(\frac{P(y=1)}{P(y=0)}\right) = a + Xb + e$$
+
+In this case, the $y$ is the logged odds of the event occurring (in the
+challenge the event was disease). In order to interpret the model results, we
+need to exponentiate both sides of the equation.
+
+$$ \exp\left(\ln\left(\frac{p}{1 - p}\right)\right) = \exp(a + Xb + e)$$
+$$ \frac{p}{1 - p} = e^{a + Xb + e}$$
+
+where $p$ is the probability of the event (equal to 1, in this case of disease).
+So if we exponentiate the estimates, and solve for the formula, we can interpret
+the result as the odds of an event occurring.
+
+```{r}
+exp(-1.035 + (-0.0003 * 50) + (-0.0338 * 0) + (-0.0004 * 30))
+```
+
+In this case, for a female with a weight of 50 and hindfoot of 30, the odds of
+disease is 0.345, fairly small. Interpretation is actually a bit more
+complicated than that. So depending on your final project, if you need/want to
+use this, we can help walk you through this more.
+
+## Confidence intervals, and p-values
```

Comment:
I think it would be interesting to mention effect size also. Just briefly remind them that no matter their p-value or confidence interval, it is also important to assess whether the observed differences in the measurement could lead to actual biological differences in the process the study.